File: src/main/java/com/matamercer/microblog/web/api/v1/dto/responses/PostResponseDto.java
package com.matamercer.microblog.web.api.v1.dto.responses;

import com.matamercer.microblog.models.entities.Blog;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class PostResponseDto extends BaseModelResponseDto {

    private BlogResponseDto blog;

    private String content;

    private String title;

    private List<PostTagResponseDto> postTags;

    private List<FileResponseDto> attachments;

    private List<LikeResponseDto> likes;

    private List<PostResponseDto> replies;

    private boolean communityTaggingEnabled;

    private boolean sensitive;

    private boolean published;

    private String parentPostId;

}

File: src/main/java/com/matamercer/microblog/web/api/v1/dto/responses/PostResponseDto.kt
package com.matamercer.microblog.web.api.v1.dto.responses

class PostResponseDto : BaseModelResponseDto() {
    var blog: BlogResponseDto? = null
    var content: String? = null
    var title: String? = null
    var postTags: List<PostTagResponseDto>? = null
    var attachments: List<FileResponseDto>? = null
    var likeCount: Int = 0
    var isLiked: Boolean? = null
    var replies: List<PostResponseDto>? = null
    var communityTaggingEnabled = false
    var sensitive = false
    var published = false
    var parentPostId: String? = null
}